rotkehlchen/globaldb/upgrades/v9_v10.py

```diff
@@ -0,0 +1,73 @@
+from typing import TYPE_CHECKING
+
+from rotkehlchen.logging import enter_exit_debug_log
+from rotkehlchen.utils.progress import perform_globaldb_upgrade_steps, progress_step
+
+if TYPE_CHECKING:
+    from rotkehlchen.db.drivers.gevent import DBConnection, DBCursor
+    from rotkehlchen.db.upgrade_manager import DBUpgradeProgressHandler
+
+
+@enter_exit_debug_log(name='globaldb v9->v10 upgrade')
+def migrate_to_v10(connection: 'DBConnection', progress_handler: 'DBUpgradeProgressHandler') -> None:  # noqa: E501
+    """This globalDB upgrade does the following:
+
+    1. Adds main_asset column to asset_collections table.
+
+    This upgrade took place in v1.37
+    """
+
+    @progress_step('Adding main_asset column to asset_collections')
+    def add_main_asset_column(write_cursor: 'DBCursor') -> None:
+        write_cursor.execute('CREATE TABLE multiasset_mappings_backup AS SELECT * FROM multiasset_mappings')  # noqa: E501
+        write_cursor.execute("""
+            CREATE TABLE asset_collections_new(
+                id INTEGER PRIMARY KEY,
+                name TEXT NOT NULL,
+                symbol TEXT NOT NULL,
+                main_asset TEXT NOT NULL UNIQUE,
+                FOREIGN KEY(main_asset) REFERENCES assets(identifier) ON UPDATE CASCADE ON DELETE CASCADE,
+                UNIQUE(name, symbol)
+            );
+        """)  # noqa: E501
+        write_cursor.execute("""
```

Comment:
This seems unnecessarily complicated as an sql statmenent. Would it not be simpler and easier to read if you used python and memory too? There is no need to do it all in one sqlite execute.

If it works good, but it's complicated. And that means error prone.